### PR TITLE
[BEAM-1695] Improve Python-SDK's programming guide

### PR DESCRIPTION
**Pull request by tibkiss**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by tibkiss** [comment]
R: @aaltay

**Comment by tibkiss** [comment]
Please make sure that this gets reviewed & merged together with the respective beam-site change:
https://github.com/apache/beam-site/pull/225

I did my best to test it locally (by running a local site, replacing the github references to local), but I recommend checking the website after the upgrade to ensure correctness.

File: sdks/python/apache_beam/examples/snippets/snippets.py [hunk 1]
```diff
@@ -155,6 +161,7 @@ def _add_argparse_args(cls, parser):
 
 def model_pcollection(argv):
   """Creating a PCollection from data in local memory."""
+  # [START model_pcollection]
```
Comment (aaltay) [inline sdks/python/apache_beam/examples/snippets/snippets.py]: I do not think this needs to go up and include the definition for options to show how `Create` works.

Comment (tibkiss) [inline sdks/python/apache_beam/examples/snippets/snippets.py]: Without the definition of options my_options.output will be undefined, which is referenced by WriteToText.

If we don't want to include MyOptions then we should replace WriteToText with some predefined filename. That case the Java & Python code will be somewhat different.

Comment (aaltay) [inline sdks/python/apache_beam/examples/snippets/snippets.py]: How about cutting the snippet from the bottom and removing:
```
 | beam.io.WriteToText(my_options.output))

result = p.run()
```
Writing and running are not part of the Java snippet.

Comment (tibkiss) [inline sdks/python/apache_beam/examples/snippets/snippets.py]: Good point, will update the PR accordingly!

**Comment by aaltay** [comment]
LGTM

**Pull request closed without merging**